### ISSUE TITLE
learn rate question

### ISSUE THREAD

**Issue by LILILILILIYIRAN** [koyeongmin/PINet_new#17]
I have some question for trim. I would appreciate it if you could answer me.
If I comment out trim of line 297 for Tusample/agent.py
How do I handle those trim of 300, 304, 306......
I think the learning rate and decay value has been given in parameter.py. Can I set trim=0 in line 297 or remove other lines trim directly.
Because when I did not comment out trim on line 297. I got a different 804_tensor result. My accuracy is not high.
If I comment out trim in line 297 as authors changed, I'll get some bugs about "not finding trim value".
I would be grateful if you could answer my question.
Thanks.

**Comment by LILILILILIYIRAN** [comment]
I also read the author's article. There is no detailed section on the parameters used for learning rate. Can you provide some detailed rules about learning rate and weight-decay?

**Comment by koyeongmin** [comment]
"agent.py" defines the change of learning rate according to epoch. If you try to train the model from scratch, please set trim=0 initially. Also, 804 is the exact epoch to get maximum performance because the label is little changed to experiment with lots of cases when I wrote the paper. In my case, I got the highest performance at about 100~200 epoch. Thank you!

**Comment by LILILILILIYIRAN** [comment]
Thanks. I have another question. How to modify the learning rate if I want to train from scratch. Modify the parameters in the parameter.py file or modify the parameters in the agent.py file.

**Comment by koyeongmin** [comment]
In parameter.py, you can modify the learning rate. Thank you!

**Comment by LILILILILIYIRAN** [comment]
Thank you for your reply. If I set trim=0, the learning rate starts at 0.0005. But in parameter.py, the value of learning rate is 0.00001.Which one should I use? If you  can tell me I will be grateful.

**Comment by koyeongmin** [comment]
In my case, I set the initial learning rate to 0.001 (in parameters.py). After 30 epoch, the learning rate will be changed to 0.005. Thank you!

**Comment by LILILILILIYIRAN** [comment]
Thanks for your reply. But when I do some experiment. I meet a mistake.  I change to the 3H and get some result. Training is good enough, but there was a problem with the evaluation. In train.py 110 lins and test.py 105, 271 lins. They told me that in test.py 271 lins, list index out of the range. I would appreciate it if the author could tell me how to solve this problem.

**Comment by koyeongmin** [comment]
How did you modify "network.py" for changing original code to 3H?

**Comment by LILILILILIYIRAN** [comment]
I've commented  <self.layer4 = hourglass_block(128, 128)> and <result4, out, feature4 = self.layer4(out)> , also I modify the <return [result1, result2, result3, result4], [feature1, feature2, feature3, feature4]> to <return [result1, result2, result3], [feature1, feature2, feature3]>, is it right? Do I need to modify any other files?

**Comment by koyeongmin** [comment]
In train.py, the code evaluates all case of hourglass (1H~4H). So if you are using only three hourglass, you need to modify line 105 in train.py. (from "index=[3]" to "index=[2]" )
